### PR TITLE
Group profile gatekeeper

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: mod/groups/start.php [hunk 1]
```diff
@@ -1197,6 +1198,31 @@ function _groups_gatekeeper_redirect(\Elgg\Hook $hook) {
...
+	if (!$exception instanceof \Elgg\GroupGatekeeperException) {
+		return;
+	}
+	
+	$group = $hook->getEntityParam();
+	if (!$group instanceof ElggGroup) {
+		return;
+	}
+	
+	if (!elgg_in_context('group_profile')) {
```
Comment (hypeJunction) [inline mod/groups/start.php]: We should move away from context sniffing. Comparing against current route name would be better

Comment (jeabakker) [reply mod/groups/start.php]: So compare current_url to group route?

Comment (hypeJunction) [reply mod/groups/start.php]: Let's of that for now. Will add public API to compare routes later

Comment (jeabakker) [reply mod/groups/start.php]: > Will add public API to compare routes later

#11671 for reference

Comment (jeabakker) [reply mod/groups/start.php]: changed it

**Pull request merged**